datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -593,15 +593,8 @@ impl LogicalPlanBuilder {
 
     /// Apply a union, removing duplicate rows
     pub fn union_distinct(self, plan: LogicalPlan) -> Result<Self> {
-        // unwrap top-level Distincts, to avoid duplication
```

Comment:
The key point of this PR is that reshuffling around of `DISTINCT` should be done as an optimization pass, rather than as part of the builder, if I understand it. This makes sense to me